metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -2016,6 +2021,70 @@ ListPartitionReassignmentsResponseData listPartitionReassignments(
         return response;
     }
 
+    ControllerResult<AssignReplicasToDirsResponseData> handleAssignReplicasToDirs(AssignReplicasToDirsRequestData request) {
```

Comment:
We should check if the Metadata version supports this request and return an error if not.

Reply:
I've added a check in `ControllerApis.handleAssignReplicasToDirs` similar to `handleDescribeCluster`.